pkgs/development/haskell-modules/default.nix

```diff
@@ -18,7 +19,8 @@ let
       mkDerivation = pkgs.callPackage ./generic-builder.nix {
         inherit stdenv;
         inherit (pkgs) fetchurl pkgconfig glibcLocales coreutils gnugrep gnused;
-        inherit (self) ghc jailbreak-cabal;
+        inherit (self) ghc;
+        jailbreak-cabal = if jailbreak-cabal != null then jailbreak-cabal else self.jailbreak-cabal;
```

Comment:
We already have two mechanisms to replace `jailbreak-cabal`: the value can be overridden for the entire package set, and it can be overridden directly on `mkDerivation`. Adding a third way seems superfluous to me.